### PR TITLE
[FVM][Execution] Adding metrics for computation effort vector

### PR DESCRIPTION
**Pull request by Tonix517**
Now emitting the raw(unweighted) computation intensities at per-block level, for sampled transactions only. So even if there are many items in the computation effort vector, we don't expect there'll be obvious extra overhead for network traffic.

Also, we don't apply `stringer` on the constants of `ComputationKind` [defined](https://github.com/onflow/flow-go/blob/master/fvm/environment/meter.go#L14) in flow-go:
- [stringer](https://pkg.go.dev/golang.org/x/tools/cmd/stringer) does not support generating string() for the cross-module constant values - in our case the type `ComputationKind` is defined in Cadence, so unless we define a new type in fvm, stringer won't generate String() for FVM's values.
- Instead we utilize Grafana override feature to map certain value like `ComputationKind(2020)` to the actual command string manually.

**Test**
- [X] show dashboard for localnet 
unweighted_tx_transx

### CODE REVIEW

File: engine/execution/messages.go [hunk 1]
```diff
@@ -17,18 +18,19 @@ type ComputationOrder struct {
...
+	ExecutableBlock        *entity.ExecutableBlock
+	StateSnapshots         []*delta.SpockSnapshot
+	StateCommitments       []flow.StateCommitment
+	Proofs                 [][]byte
+	Events                 []flow.EventsList
+	EventsHashes           []flow.Identifier
+	ServiceEvents          flow.EventsList
+	TransactionResults     []flow.TransactionResult
+	ComputationUsed        uint64
+	ComputationIntensities meter.MeteredComputationIntensities
```
Comment (janezpodhostnik) [inline engine/execution/messages.go:30]: Can you check if these (+memory) could be packaged into their own struct called `MeteringResults`?

File: fvm/transactionInvoker.go [hunk 2]
```diff
@@ -171,6 +171,9 @@ func (i TransactionInvoker) Process(
 	proc.Logs = append(proc.Logs, env.Logs()...)
 	proc.ComputationUsed = proc.ComputationUsed + env.ComputationUsed()
 	proc.MemoryEstimate = proc.MemoryEstimate + env.MemoryEstimate()
+	if proc.IsSampled() {
+		proc.ComputationIntensities = env.ComputationIntensities()
+	}
```
Comment (janezpodhostnik) [inline fvm/transactionInvoker.go:176]: maybe wrap these in a function.

File: engine/execution/computation/computer/computer.go [hunk 3]
```diff
@@ -456,6 +458,9 @@ func (e *blockComputer) executeTransaction(
 	res.AddServiceEvents(tx.ServiceEvents)
 	res.AddTransactionResult(&txResult)
 	res.AddComputationUsed(tx.ComputationUsed)
+	if tx.IsSampled() {
```
Comment (janezpodhostnik) [inline engine/execution/computation/computer/computer.go:461]: This condition probably isn't necessary. Just make sure `MergeComputationEffortVector` can handle nil values.

**Review by janezpodhostnik — APPROVED** [review]
Some suggestions.

File: engine/execution/messages.go [hunk 4]
```diff
@@ -47,6 +49,13 @@ func (cr *ComputationResult) AddComputationUsed(inp uint64) {
 	cr.ComputationUsed += inp
 }
 
+func (cr *ComputationResult) MergeComputationEffortVector(
+	computationIntensities meter.MeteredComputationIntensities) {
```
Comment (pattyshack) [inline engine/execution/messages.go:53]: add newline to clarify end of method signature.

**Comment by Tonix517** [comment]
for clarity i'll address your comments in a separate PR specifically for refactoring.

**Comment by Tonix517** [comment]
bors merge

**Comment by Tonix517** [comment]
after [discussion](https://docs.google.com/document/d/1b0HtraCXs57o44JexkbidXj4F-iAlhPuVFj7NsCrNPc/edit?usp=sharing) we decided to use state-replay cache to do cached state reads while keeping the original reg touch records.

**Comment by Tonix517** [comment]
mistook it with #3240

**Comment by Tonix517** [comment]
bors merge

**Pull request merged**